apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainTest.java

```diff
@@ -924,45 +924,45 @@ void testGetTransactionsWithConfirmation() {
 
         List<Transaction> transactions = blockchain.getTransactions(txd.TRANSACTION_2.getSenderId(), txd.TRANSACTION_14.getHeight() - txd.TRANSACTION_8.getHeight() + 1, (byte) -1, (byte) -1, 0, false, false, false, 0, Integer.MAX_VALUE, false, false, true);
 
-        assertEquals(List.of(txd.TRANSACTION_6, txd.TRANSACTION_5, txd.TRANSACTION_4, txd.TRANSACTION_3, txd.TRANSACTION_2, txd.TRANSACTION_1, txd.TRANSACTION_0), transactions);
+        assertEquals(List.of(txd.TRANSACTION_6, txd.TRANSACTION_5, txd.TRANSACTION_4, txd.TRANSACTION_3, txd.TRANSACTION_2, txd.TRANSACTION_1), transactions);
     }
 
     @Test
     void testGetTransactionsFromDifferentDataSources() {
         blockchain.setLastBlock(btd.BLOCK_13);
 
-        List<Transaction> transactions = blockchain.getTransactions(txd.TRANSACTION_2.getSenderId(), 0, (byte) -1, (byte) -1, 0, false, false, false, 0, Integer.MAX_VALUE, false, false, true);
-
-        assertEquals(List.of(txd.TRANSACTION_13, txd.TRANSACTION_12, txd.TRANSACTION_11, txd.TRANSACTION_9, txd.TRANSACTION_8, txd.TRANSACTION_7, txd.TRANSACTION_6, txd.TRANSACTION_5, txd.TRANSACTION_4, txd.TRANSACTION_3, txd.TRANSACTION_2, txd.TRANSACTION_1, txd.TRANSACTION_0), transactions);
+        List<Transaction> transactions = blockchain.getTransactions(txd.TRANSACTION_2.getSenderId(), 0, (byte)8, (byte)0, 0, false, false, false, 0, Integer.MAX_VALUE, false, false, true);
```

Comment:
Could you please add additional test case, where bug with retrieving transactions can be reproduced? For now, i see only edited tests.

Reply:
Bug is reproducible in previous commit as I wrote in chat.

The main unit test is following.
For given accountId = txd.TRANSACTION_2.getSenderId()
we should fetch all transaction records by type/sub-type = 8, 0
from main + shard1 + shard2 databases, because all those records are split among all those files.